### PR TITLE
23942 - If transaction is already completed jump directly to redirect url

### PR DESCRIPTION
**Pull request by seeker25**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by seeker25** [comment]
I'm seeing security implications due to this, have to think about it before merging

**Comment by seeker25** [comment]
Needs https://github.com/bcgov/sbc-pay/pull/1794

**Review by ochiu — APPROVED** [review]
Looks good.

File: auth-web/src/views/pay/PaymentView.vue [hunk 1]
```diff
@@ -238,11 +240,15 @@ export default class PaymentView extends Vue {
...
   }
 
-  private doHandleError (error) {
+  async doHandleError (error) {
     this.showLoading = false
     this.errorMessage = this.$t('payFailedMessage').toString()
-    if (error.response.data && error.response.data.type === 'INVALID_TRANSACTION') {
-      // Transaction is already completed. Show as a modal.
+    if (error.response.data && ['COMPLETED_PAYMENT', 'INVALID_TRANSACTION'].includes(error.response.data.type)) {
+      // Skip PAYBC, take directly to the "clients redirect url", this avoids transaction already done error.
```
Comment (seeker25) [inline auth-web/src/views/pay/PaymentView.vue:247]: image

**Pull request merged**